flang/include/flang/Optimizer/Dialect/FIRTypes.td

```diff
@@ -21,6 +21,32 @@ class FIR_Type<string name, string typeMnemonic> : TypeDef<fir_Dialect, name> {
   let mnemonic = typeMnemonic;
 }
 
+def fir_ComplexType : FIR_Type<"Complex", "complex"> {
+  let summary = "Complex type";
+
+  let description = [{
+    Model of a Fortran COMPLEX intrinsic type, including the KIND type
+    parameter. COMPLEX is a floating point type with a real and imaginary
+    member.
+  }];
+
+  let parameters = (ins "KindTy":$kind);
```

Comment:
Were you going to rename this `fKind`?

Reply:
I can update it if it makes more sense so there will be only one accessor in the end. What do you think?

Reply:
Makes sense to just have one.

Reply:
Ok I'll update it.